### ISSUE TITLE
ClientAnimator crash [Primitive Survival Bioluminescent Creature]

### ISSUE THREAD

**Issue by WickedSchnitzel** [anegostudios/VintageStory-Issues#4243]
### Game Version

1.20-pre5

### Platform

Windows

### Modded

Modded

### SP/MP

Multiplayer

### Description

Not sure if this is mod related maybe. I did remove mods with animated entities from the server to test and it still happened.

### Logs

```System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.API.Common.ClientAnimator.calculateMatrices(Single dt) in VintagestoryApi\Common\Model\Animation\ClientAnimator.cs:line 294
   at Vintagestory.API.Common.AnimatorBase.OnFrame(Dictionary`2 activeAnimationsByAnimCode, Single dt) in VintagestoryApi\Common\Model\Animation\AnimatorBase.cs:line 109
   at Vintagestory.API.Common.AnimationManager.OnClientFrame(Single dt) in VintagestoryApi\Common\Model\Animation\AnimationManager.cs:line 430
   at Vintagestory.Client.NoObf.SystemRenderEntities.OnBeforeRender(Single dt) in VintagestoryLib\Client\Systems\Render\RenderEntities.cs:line 74
   at Vintagestory.Client.NoObf.ClientEventManager.TriggerRenderStage(EnumRenderStage stage, Single dt) in VintagestoryLib\Client\Util\ClientEventManager.cs:line 186
   at Vintagestory.Client.NoObf.ClientMain.TriggerRenderStage(EnumRenderStage stage, Single dt) in VintagestoryLib\Client\ClientMain.cs:line 818
   at Vintagestory.Client.NoObf.ClientMain.MainRenderLoop(Single dt) in VintagestoryLib\Client\ClientMain.cs:line 824
   at Vintagestory.Client.NoObf.ClientMain.MainGameLoop(Single deltaTime) in VintagestoryLib\Client\ClientMain.cs:line 740
   at Vintagestory.Client.GuiScreenRunningGame.RenderToPrimary(Single dt) in VintagestoryLib\Client\MainMenu\Screens\GuiScreenRunningGame.cs:line 207
   at Vintagestory.Client.ScreenManager.Render(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 695
   at Vintagestory.Client.ScreenManager.OnNewFrame(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 670
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(FrameEventArgs e) in VintagestoryLib\Client\ClientPlatform\GameWindow.cs:line 88
   at OpenTK.Windowing.Desktop.GameWindow.Run()
   at Vintagestory.Client.ClientProgram.Start(ClientProgramArgs args, String[] rawArgs) in VintagestoryLib\Client\ClientProgram.cs:line 328
   at Vintagestory.Client.ClientProgram.<>c__DisplayClass10_0.<.ctor>b__1() in VintagestoryLib\Client\ClientProgram.cs:line 130
   at Vintagestory.ClientNative.CrashReporter.Start(ThreadStart start) in VintagestoryLib\Client\ClientPlatform\ClientNative\CrashReporter.cs:line 93```

**Comment by maltiez2** [comment]
@WickedSchnitzel please send client-main and server-main logs after crash

**Comment by iSoulEs** [comment]
> lacking more data

This is still happening even in v1.20-rc.4

logs:
[client-main.log](https://github.com/user-attachments/files/18158433/client-main.log)

[server-main.log](https://github.com/user-attachments/files/18158450/server-main.log)

**Comment by AreYSerious** [comment]
https://www.youtube.com/watch?v=VNC2Ct4hovc

related?

**Comment by AreYSerious** [comment]
@Craluminum2413 

might wana reopen

**Comment by AreYSerious** [comment]
I think its not a real vanilla issue but its not something mods can fix.

Is this some kind of shape overload? too many chiseled blocks / clutter blocks

**Comment by AreYSerious** [comment]
https://github.com/AreYSerious/betterruins/issues/51

**Comment by AreYSerious** [comment]
I have a 13GB world with a place to get it to crash consistently if thats needed let me know

**Comment by AreYSerious** [comment]
plently mods tho

**Comment by radfast** [comment]
@AreYSerious @WickedSchnitzel  please post relevant client-main.log showing the full modlist as well as the crash, or at least **post your modlist** when this occurred.

In comparison, @iSoulEs posted recent logs (thank you) including these mods:

      Loaded Mods: decor@1.2.0, juicyores@1.0.0, TA_plainsandvalleys@1.0.9, primitivesurvival@3.7.4, proximityprospectingpick@1.0.2, game@1.20.0-rc.4, BetterProspecting@1.3.0, butchering@1.7.3, commonlib@2.6.1, danatweaks@3.3.7, extrainfo@1.8.2, firewoodtosticks@1.0.0, creative@1.20.0-rc.4, survival@1.20.0-rc.4, wildfarmingrevival@1.3.0, helvehammerext@1.7.0

As a starting point it would be good to see if all three people reporting happen to be using 1 mod in common

**Comment by AreYSerious** [comment]
https://github.com/AreYSerious/betterruins/issues/51

Did some testing here but it's very random and not easy to replicate besides with the full modlist on the save where I did the video on.

It's very random and seems to not trigger consistently.

**Comment by AreYSerious** [comment]
I have played around with removing mods and trying to find out if it's a specific mod but when I disabled some it wasn't really consistently replicateable so I stopped. It just didn't crash for me in a way that I could say a specific group of mods or one mod would cause this. But maybe it's possible with a bit more testing.

@radfast could you give us a bit more information about the crash? What is causing it how to replicate it. What should we do? I was guessing just looking at different clutter/chiseled blocks might cause it? Bcs it's related to shapes? No idea.


Pls check the issue on BetterRuins repo there are some crash logs there

**Comment by AreYSerious** [comment]
I can also give you a savefile with modlist 1:1 and also location in the map which should allow to replicate it. But only after Christmas so prob in 2-3 days

**Comment by WickedSchnitzel** [comment]
Loaded Mods: ancientarmory@1.0.6, bettercrates@1.8.0-rc.4, canoemod@1.0.2, chiseltools@1.14.4, decor@1.2.0, extrachests@1.8.0, fantasycreatures@0.7.1, fixedpaths@1.0.1, pantherinae@1.1.14, foundrymod@1.0.0, moreicons_cs@1.1.0, primitivesurvival@3.7.4, nvscaffolding@1.0.1, translocatorengineeringredux@1.5.4, game@1.20.0-rc.6, weaponpackalphaunoff@1.5.1, animalcages@3.0.3, apegrapes@1.1.3, betterruins@0.4.2, butterflycases@1.0.1, carryon@1.8.0-pre.1, commonlib@2.6.1, decayingcreatures@0.1.1, fairplayguardian@1.1.7, furniturelibrary@1.0.14, gourmand@1.2.7, heatretention@1.0.2, playerlist@2.1.5, rifteye@0.4.0, simplefootstepsredux@1.0.1, sleepvote@1.0.7, tentbag@2.2.0, th3dungeon@0.3.1, thecritterpack@0.9.1, vanity@2.2.1, vinconomy@0.4.0-rc.4, creative@1.20.0-rc.6, vsquest@1.2.0, vsserverpanel@0.1.5, survival@1.20.0-rc.6, wildgrass@1.1.3, youvegotmail@1.0.8, cob2@1.6.9, em@3.0.0-pre.3, outerspacetunesvolume1@1.0.2, outerspacetunesvolume2@1.0.2, playercorpse@1.11.0, stonequarry@3.4.2, cob@2.0.0, outerspacetunesmaster@1.0.2

**Comment by WickedSchnitzel** [comment]
decor, primitivesurvival and commonlib are used by both of us, from what i see

**Comment by AreYSerious** [comment]
https://youtu.be/VNC2Ct4hovc



```
Game Version: v1.20.0-rc.5 (Unstable)
20.12.2024 23:06:12: Critical error occurred
Loaded Mods: advancedsmithing@1.3.0, aged@1.0.6, ancientarmory@1.0.6, arrowheadsrecovery@1.0.1, bettercrates@1.8.0-rc.4, betterloot@1.0.2, scrapblocks@1.1.0, bettershears@1.0.0, brickmold@1.2.0, canoemod@1.0.2, chiseltools@1.14.3, claycasting@1.2.2, darcesdrifters@1.0.0, easyelk@1.0.0, extrachests@1.8.0, fancysky@1.1.6, hideplayername@2.1.1, liberatedbeds@1.0.0, longtermfood@0.1.0, millwright@1.1.8, molds@0.1.3, nailsmold@1.0.2, oils@1.2.3, primitivesurvival@3.7.4, ripened_wild_crops@1.0.0, sortablestorage@2.2.8, stepfixelk@0.0.1, temporal_gears_stack@1.0.0, terraprety@5.0.5, translocatorengineeringredux@1.5.4, visibleore@1.0.1, game@1.20.0-rc.5, alchemy@1.6.41, animalcages@3.0.3, apegrapes@1.1.3, egocaribautomapmarkers@4.0.0, barreldough@2.0.0, betterfirepit@1.1.4, betterhewnfencegates@1.1.0, betterruins@0.4.1, butchering@1.7.3, carryon@1.7.4, commonlib@2.6.1, configureeverything@2.1.3, cooperativecombat@1.1.1, Crateful@1.2.8, danacancook@0.2.4, danatweaks@3.3.7, extrainfo@1.9.2, farmlanddropssoil@1.4.0, fixhandbookclutter@1.0.5, foodshelves@1.4.4, fromgoldencombs@1.8.1, gourmand@1.2.6, labeledtrunk@1.0.5, playerlist@2.1.4, claywheel@1.1.2, prospecttogether@1.4.0, rivers@3.2.0, scarecrow@1.6.1, stepupcontinued@0.0.1, usefulstuff17patch@1.0.0, stonebakeoven@1.1.4, storagecontroller@1.1.2, vanity@2.2.1, creative@1.20.0-rc.5, vsimgui@1.1.5, survival@1.20.0-rc.5, waypointtogether@1.0.1, wildfarmingrevival@1.2.9, configlib@1.3.13, dowsedinsalt@1.0.0, helvehammerext@1.7.0, playercorpse@1.11.0, stonequarry@3.4.2, thriftysmithing@1.2.0, statushudcont@2.1.2
System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.API.Common.ClientAnimator.calculateMatrices(Single dt) in VintagestoryApi\Common\Model\Animation\ClientAnimator.cs:line 294
   at Vintagestory.Client.NoObf.SystemRenderEntities.OnBeforeRender(Single dt) in VintagestoryLib\Client\Systems\Render\RenderEntities.cs:line 49
   at Vintagestory.Client.NoObf.ClientEventManager.TriggerRenderStage(EnumRenderStage stage, Single dt) in VintagestoryLib\Client\Util\ClientEventManager.cs:line 188
   at Vintagestory.Client.NoObf.ClientMain.MainRenderLoop(Single dt) in VintagestoryLib\Client\ClientMain.cs:line 824
   at Vintagestory.Client.GuiScreenRunningGame.RenderToPrimary(Single dt) in VintagestoryLib\Client\MainMenu\Screens\GuiScreenRunningGame.cs:line 161
   at Vintagestory.Client.ScreenManager.Render(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 695
   at Vintagestory.Client.ScreenManager.OnNewFrame(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 670
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(FrameEventArgs e) in VintagestoryLib\Client\ClientPlatform\GameWindow.cs:line 92
   at OpenTK.Windowing.Desktop.GameWindow.Run()
   at Vintagestory.Client.ClientProgram.Start(ClientProgramArgs args, String[] rawArgs) in VintagestoryLib\Client\ClientProgram.cs:line 334
   at Vintagestory.Client.ClientProgram.<>c__DisplayClass10_0.<.ctor>b__1() in VintagestoryLib\Client\ClientProgram.cs:line 130
   at Vintagestory.ClientNative.CrashReporter.Start(ThreadStart start) in VintagestoryLib\Client\ClientPlatform\ClientNative\CrashReporter.cs:line 93
```

**Comment by radfast** [comment]
Right so all three reporting this are using `primitivesurvival@3.7.4`.   I would start by looking there.

It's difficult for us to tell precisely which line of code is crashing, as in 1.20 the runtime code makes additional optimisations the longer the player plays, and those optimisations (annoyingly) change the line numbers if a stacktrace is then reported.

At a guess, either one of a creature's currently running animations is null (which would most likely be an engine bug) or there is an animation which has one or more fields missing (which would be a mod bug).  But any engine bug with the animations systems would likely be triggered by Survival, as 1.20 includes some very heavily animated entities including the new mobs.

Unless someone who is _not_ running `primitivesurvival` reports the same issue, I think the evidence so far indicates it's more likely to be a mod issue.  _Provisionally_ therefore I will close this, but we can re-open if necessary

**Comment by SpearAndFang** [comment]
While I wouldn't be surprised...there's no Primitive Survival here fyi
https://github.com/anegostudios/VintageStory-Issues/issues/4605

**Comment by AreYSerious** [comment]
@Craluminum2413 reopen again please 🎉😅

**Comment by radfast** [comment]
> While I wouldn't be surprised...there's no Primitive Survival here fyi #4605

Ah, @SpearAndFang forgive me I didn't check the duplicate issue.  So not Primitive Survival, then.

**Comment by radfast** [comment]
> I can also give you a savefile with modlist 1:1 and also location in the map which should allow to replicate it. But only after Christmas so prob in 2-3 days

Yes please, when you're ready if you can give us the following:

* savefile
* zip of the /Mods folder
* map location

then I think one of the dev team should go there, try to reproduce the issue. If we can, then we can use breakpoints in the code to find the root cause.

**Comment by AreYSerious** [comment]
@radfast awesome thank you. I will give you all that tomorrow when I'm home.

**Comment by digital-r** [comment]
Oh, i believe, primitivesurvivals will-o-wisps can be responsible for this crash. Some time ago, i have numerous crashes in areas with high populations of this "rare" creatures. Was fixed by adjusting maxanimatedelements client setting, iirc.

**Comment by SpearAndFang** [comment]
anecdotal, but I also have will-o-wisps in my area (about 30 hours of gameplay) and no crashes.
and for what it's worth. my setting:

  "maxAnimatedElements": 230,

**Comment by digital-r** [comment]
If it can help, i have much lower value for maxanimatedelements on that notebook - 48 (was 24 before adjustment, iirc).

But i never have such crashes on my main pc in this place. (With maxanimatedelements set to 120).

**Comment by radfast** [comment]
> If it can help, i have much lower value for maxanimatedelements on that notebook - 48 (was 24 before adjustment, iirc).
> 
> But i never have such crashes on my main pc in this place. (With maxanimatedelements set to 120).

Vanilla goats, and in particular Musk Ox, will crash the game if MaxAnimatedElements is set as low as 48, essentially they were the reason the default was increased to 56 in 1.19.  In 1.20 the cap has been greatly lifted.  MaxAnimatedElements is not likely to be related to the crashes reported in this thread, I think.

For now I am keeping an open mind on the root cause.

**Comment by digital-r** [comment]
Looks like, game never update this settings in clientsettings file, when new version installed. At least, when client reinstalled with authomatic deinstallation of previous game version.

**Comment by AreYSerious** [comment]
https://1drv.ms/f/s!AvmodXdmrW-Qj441V4pELIb5F627mA?e=pI5qdC

OneDrive Folder which will contain the Data folder.

If you join the world 

/tpstoryloc university

to get to the location

go to the tower like you saw in the youtube video and go into survival and then look at all kinds of clutter blocks and chiseled blocks after a time it will crash.

https://youtu.be/VNC2Ct4hovc

**Comment by Deoxys0** [comment]
I'm having the same issue; running a lot more mods than the original user but it doesn't seem to be a mod issue.
`Game Version: v1.20.0-rc.6 (Unstable)
12/27/2024 6:33:31 PM: Critical error occurred
Loaded Mods: ancientarmory@1.0.6, floralzonescaperegion@1.0.13, floralzonescaribbeanregion@1.0.10, floralzonescentralaustralianregion@1.0.7, floralzoneseastasiaticregion@1.0.5, floralzonesmediterraneanregion@1.0.2, floralzonesneozeylandicregion@1.0.6, caninae@1.0.22, capreolinae@1.1.11, casuariidae@1.0.14, dinornithidae@1.0.5, machairodontinae@1.0.15, manidae@1.0.7, pantherinae@1.1.15, rhinocerotidae@1.0.9, sirenia@1.0.10, spheniscidae@1.0.2, hangingoillamps@1.0.2, millwright@1.1.8, primitivesurvival@3.7.4, temporal_gears_stack@1.0.0, translocatorengineeringredux@1.5.4, game@1.20.0-rc.6, vsimgui@1.1.7, betterruins@0.4.2, canjewelry@0.4.5, chestorganizer@1.0.3, commonlib@2.6.1, configlib@1.3.17, fendragonbcs@0.1.3, fromgoldencombs@1.8.1, hudclockpatch@1.0.0, knapster@2.11.2, prospecttogether@1.4.0, spawnhighlight@1.2.0, th3dungeon@0.3.1, creative@1.20.0-rc.6, survival@1.20.0-rc.6, wildfarmingrevival@1.3.1, helvehammerext@1.7.0, stonequarry@3.4.2
System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.API.Common.ClientAnimator.calculateMatrices(Single dt) in VintagestoryApi\Common\Model\Animation\ClientAnimator.cs:line 294
   at Vintagestory.Client.NoObf.SystemRenderEntities.OnBeforeRender(Single dt) in VintagestoryLib\Client\Systems\Render\RenderEntities.cs:line 49
   at Vintagestory.Client.NoObf.ClientEventManager.TriggerRenderStage(EnumRenderStage stage, Single dt) in VintagestoryLib\Client\Util\ClientEventManager.cs:line 188
   at Vintagestory.Client.NoObf.ClientMain.MainRenderLoop(Single dt) in VintagestoryLib\Client\ClientMain.cs:line 825
   at Vintagestory.Client.GuiScreenRunningGame.RenderToPrimary(Single dt) in VintagestoryLib\Client\MainMenu\Screens\GuiScreenRunningGame.cs:line 161
   at Vintagestory.Client.ScreenManager.Render(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 695
   at Vintagestory.Client.ScreenManager.OnNewFrame(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 670
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(FrameEventArgs e) in VintagestoryLib\Client\ClientPlatform\GameWindow.cs:line 92
   at OpenTK.Windowing.Desktop.GameWindow.Run()
   at Vintagestory.Client.ClientProgram.Start(ClientProgramArgs args, String[] rawArgs) in VintagestoryLib\Client\ClientProgram.cs:line 334
   at Vintagestory.Client.ClientProgram.<>c__DisplayClass10_0.<.ctor>b__1() in VintagestoryLib\Client\ClientProgram.cs:line 130
   at Vintagestory.ClientNative.CrashReporter.Start(ThreadStart start) in VintagestoryLib\Client\ClientPlatform\ClientNative\CrashReporter.cs:line 93`

**Comment by WickedSchnitzel** [comment]
This is happening every few seconds to minutes to some players. I really need to get rid of this shit. But nobody knows anything. It must hae to do with the 1.20 changes one way or another.

**Comment by WickedSchnitzel** [comment]
Its happening to people even if they are just standing around in their base.

**Comment by Deoxys0** [comment]
If it helps, I'll add some more context to when it keeps happening to me: when I'm working on my base. But the weird part is that I spent like 20 real life hours working on this base and had zero issues, then it randomly started happening with no mod changes or anything. The only thing of note is that it seems to happen more often when I'm working near my smithing room, which has a lot of chiseled blocks. I don't know if that relates to it or not, but I've seen some other posts here about chiseled blocks so I wanted to mention it. Also, if I get a crash and then load back in, then run like 200 blocks away from my base, it'll stop crashing, assuming I escape my base quick enough. It makes me think it's a drifter issue (maybe some weird combination with drifters and chiseled blocks? although nothing should be spawning in my smithing room). If I go back to the base, I can play normally for like 30 minutes to an hour before crashing again.

**Comment by AreYSerious** [comment]
@Craluminum2413

No hurries but
Could you adjust the title and remove primitive survival mention bcs this error also happens without primitive survival

**Comment by WickedSchnitzel** [comment]
Anyone also getting

[Server Error] Exception: Object reference not set to an instance of an object.
   at Vintagestory.API.Util.ArrayExtensions.Contains[T](T[] array, T value) in VintagestoryApi\Util\ArrayExtensions.cs:line 199
   at Vintagestory.GameContent.POIRegistry.GetNearestPoi(Vec3d centerPos, Single radius, PoiMatcher matcher) in VSEssentials\Systems\POIRegistry.cs:line 159
   at Vintagestory.GameContent.AiTaskSeekFoodAndEat.ShouldExecute() in VSEssentials\Entity\AI\Task\TasksImpl\AiTaskSeekFoodAndEat.cs:line 100
   at Vintagestory.GameContent.AiTaskManager.OnGameTick(Single dt) in VSEssentials\Entity\AI\Task\AiTaskManager.cs:line 214
   at Vintagestory.GameContent.EntityBehaviorTaskAI.OnGameTick(Single deltaTime) in VSEssentials\Entity\AI\Task\BehaviorTaskAI.cs:line 125
   at Vintagestory.API.Common.Entities.Entity.OnGameTick(Single dt) in VintagestoryApi\Common\Entity\Entity.cs:line 928
   at Vintagestory.API.Common.EntityAgent.OnGameTick(Single dt) in VintagestoryApi\Common\Entity\EntityAgent.cs:line 589
   at Vintagestory.Server.ServerSystemEntitySimulation.TickEntities(Single dt) in VintagestoryLib\Server\Systems\World\EntitySimulation.cs:line 331
   at Vintagestory.Server.ServerSystemEntitySimulation.OnServerTick(Single dt) in VintagestoryLib\Server\Systems\World\EntitySimulation.cs:line 143
   at Vintagestory.Server.ServerMain.Process() in VintagestoryLib\Server\ServerMain.cs:line 896

maybe its totally unreleated. 

Or lots of errors about

28.12.2024 10:00:59 [Warning] Cannot load chiseled block id mapping for rotation @ 9, 3, 21, block id game:unknown not found block registry. Will not display correctly.
28.12.2024 10:00:59 [Warning] Cannot load chiseled block id mapping for rotation @ 9, 3, 7, block id game:unknown not found block registry. Will not display correctly.

?

**Comment by Craluminum2413** [comment]
Why you remove high priority?

**Comment by radfast** [comment]
> Why you removed high priority?

You're asking me? That was removed three days ago. I wouldn't see an issue that affects only largely modded games as falling within the High Priority definition. For example, it's not something that would stop tyron releasing the next version. We might even still find it is wholly an issue in one of the mods, for example an improperly defined animation.

Having said that I am looking at the issue myself over the next hour or so, as a first step seeing if I can reproduce it in the world @AreYSerious kindly provided.   (@AreYSerious  are you aware there's a startup code issue in the Rivers mod?)

**Comment by WickedSchnitzel** [comment]
Maybe its these new insects hopping around?

**Comment by radfast** [comment]
> https://1drv.ms/f/s!AvmodXdmrW-Qj441V4pELIb5F627mA?e=pI5qdC
> 
> OneDrive Folder which will contain the Data folder.
> 
> If you join the world
> 
> /tpstoryloc university
> 
> to get to the location
> 
> go to the tower like you saw in the youtube video and go into survival and then look at all kinds of clutter blocks and chiseled blocks after a time it will crash.
> 
> https://youtu.be/VNC2Ct4hovc

I've downloaded this world and set up a local server with it, using this serverconfig and servermagicnumbers.  I've then joined the game in a 1.20.0-rc.6 client (with matching 76 mods), gone to the university location (also seen at the coordinates in the YouTube video, thank you), derped around there looking at stuff for 30 minutes now.  No crash.

I'm not sure what else to do here. I can't debug a crash which I can't reproduce on my own PC. I'll add code to next version which (a) logs the issue instead of crashing, and (b) identifies the Entity causing the crash. Very likely there will be log spam (and perhaps massive lag spikes as a result) when the issue occurs.

**Comment by radfast** [comment]
> This is happening every few seconds to minutes to some players. I really need to get rid of this shit. But nobody knows anything. It must have to do with the 1.20 changes one way or another.

Please edit the wording of this, I don't find it helpful or constructive. The dev team wants to resolve this issue. However, we cannot resolve an issue which we are unable to reproduce.

You could help by:

(a) Communicating to us (I suggest me, th3dilli and maltiez, you can reach us by DM on Discord) your own server address, any password required, and whitelisting and oping us on your server so that we can join easily - I don't want to join and spend an hour running around at spawn or building a starter house.  If we are able to join your server and reproducibly have the issue, then we can start to troubleshoot it.
(b) Figuring out if there's a particular place in the game world where your players often/always crash with this, and then we can go directly there in debug mode and see what's there.
(c) Not clogging up the issue thread with [unrelated server-side issues](https://github.com/anegostudios/VintageStory-Issues/issues/4243#issuecomment-2564329219), if you have a separate issue to report then it should have its own open Issue here.
(d) Please don't remove your modlist from crash reports, always start the report with the game version and modlist. These things are included in the crash reports for good reasons. For one thing it allows other players with potentially the same issue to identify that maybe they have mods in common with you

**Comment by WickedSchnitzel** [comment]
> > This is happening every few seconds to minutes to some players. I really need to get rid of this shit. But nobody knows anything. It must have to do with the 1.20 changes one way or another.
> 
> Please edit the wording of this, I don't find it helpful or constructive. The dev team wants to resolve this issue. However, we cannot resolve an issue which we are unable to reproduce.
> 
> You could help by:
> 
> (a) Communicating to us (I suggest me, th3dilli and maltiez, you can reach us by DM on Discord) your own server address, any password required, and whitelisting and oping us on your server so that we can join easily - I don't want to join and spend an hour running around at spawn or building a starter house. If we are able to join your server and reproducibly have the issue, then we can start to troubleshoot it. (b) Figuring out if there's a particular place in the game world where your players often/always crash with this, and then we can go directly there in debug mode and see what's there. (c) Not clogging up the issue thread with [unrelated server-side issues](https://github.com/anegostudios/VintageStory-Issues/issues/4243#issuecomment-2564329219), if you have a separate issue to report then it should have its own open Issue here. (d) Please don't remove your modlist from crash reports, always start the report with the game version and modlist. These things are included in the crash reports for good reasons. For one thing it allows other players with potentially the same issue to identify that maybe they have mods in common with you

My server is public. Just look for "Conquest of Blocks" in the server browser. I can give you creative moderator rights, so that you can just fly and stuff.

**Comment by radfast** [comment]
> Maybe its these new insects hopping around?

The crash relates to a full size creature (including hostile mobs, or the player's own seraph).  Possibly one specific animation for one creature is bugged.  Possibly there's something wrong in game code or it just goes wrong when overwhelmed by the number of mods.

Possibly it relates to only one specific action that a creature or player is taking - a player has hundreds of different animations for different actions and poses - so even if you spawn in 10 of creature X, you won't see the issue until one of them goes into a particular state.

**Comment by radfast** [comment]
> > > This is happening every few seconds to minutes to some players. I really need to get rid of this shit. But nobody knows anything. It must have to do with the 1.20 changes one way or another.
> > 
> > 
> > Please edit the wording of this, I don't find it helpful or constructive. The dev team wants to resolve this issue. However, we cannot resolve an issue which we are unable to reproduce.
> > You could help by:
> > (a) Communicating to us (I suggest me, th3dilli and maltiez, you can reach us by DM on Discord) your own server address, any password required, and whitelisting and oping us on your server so that we can join easily - I don't want to join and spend an hour running around at spawn or building a starter house. If we are able to join your server and reproducibly have the issue, then we can start to troubleshoot it. (b) Figuring out if there's a particular place in the game world where your players often/always crash with this, and then we can go directly there in debug mode and see what's there. (c) Not clogging up the issue thread with [unrelated server-side issues](https://github.com/anegostudios/VintageStory-Issues/issues/4243#issuecomment-2564329219), if you have a separate issue to report then it should have its own open Issue here. (d) Please don't remove your modlist from crash reports, always start the report with the game version and modlist. These things are included in the crash reports for good reasons. For one thing it allows other players with potentially the same issue to identify that maybe they have mods in common with you
> 
> My server is public. Just look for "Conquest of Blocks" in the server browser. I can grant you creative moderator rights, so that you can just fly around and stuff.

Thank you, if you could grant me and the other two team members I mentioned Creative rights, that would be helpful.

I don't know if I'm able to join CoB, I don't think my GPU can handle it, but I'll give it a try.

**Comment by WickedSchnitzel** [comment]
Send me a PM on Discord, once you're on.

**Comment by radfast** [comment]
> Send me a PM on Discord, once you're on.

Thank you, it may be later this evening as I need to go make dinner now. Even devs have to eat ...

**Comment by WickedSchnitzel** [comment]
And devs should also have a proper GPU.. xD

**Comment by radfast** [comment]
> And devs should also have a proper GPU.. xD

I'm sure the other devs do. I'm travelling for 10 days, with family for the holiday period, so I have 2 laptops with me, a coding laptop and a gaming laptop. Will find out shortly if it can handle CoB or not ...

**Comment by AreYSerious** [comment]
> > https://1drv.ms/f/s!AvmodXdmrW-Qj441V4pELIb5F627mA?e=pI5qdC
> > 
> > OneDrive Folder which will contain the Data folder.
> > 
> > If you join the world
> > 
> > /tpstoryloc university
> > 
> > to get to the location
> > 
> > go to the tower like you saw in the youtube video and go into survival and then look at all kinds of clutter blocks and chiseled blocks after a time it will crash.
> > 
> > https://youtu.be/VNC2Ct4hovc
> 
> I've downloaded this world and set up a local server with it, using this serverconfig and servermagicnumbers.  I've then joined the game in a 1.20.0-rc.6 client (with matching 76 mods), gone to the university location (also seen at the coordinates in the YouTube video, thank you), derped around there looking at stuff for 30 minutes now.  No crash.
> 
> I'm not sure what else to do here. I can't debug a crash which I can't reproduce on my own PC. I'll add code to next version which (a) logs the issue instead of crashing, and (b) identifies the Entity causing the crash. Very likely there will be log spam (and perhaps massive lag spikes as a result) when the issue occurs.

This is just a world I got forwarded and they use a lot of mods that are mostly also not updated to latest game versions. But this bug is sadly a thing all random kinds of occasions.

Very weird you couldn't replicate it bcs while being quite random it worked for me multiple times and even like 3 times in 2min where I was recording the video.

**Comment by AreYSerious** [comment]
https://www.youtube.com/watch?v=hVaTUhQ8vQI

**Comment by WickedSchnitzel** [comment]
radfast, if you want to join our server discord and talk to the people in the bugs section of VS, heres the link: https://dsc.gg/thecobservers

**Comment by Kuzmi75** [comment]
Hi!

I hope this can help, I uploaded a couple of logs to [dropbox](https://www.dropbox.com/scl/fo/dvhrlbqok28k4kn5ppiut/AEp133aSi8POo-2s13IG5cM?rlkey=qd0hhdq8t0wc0focsknv1nmhp&st=ru2myoq3&dl=0), containing recent freezes on the CoB server. 
In the most varied of situations, I've had to do literally nothing at all..

**Comment by radfast** [comment]
@AreYSerious as written already the crash origin is in entity animations, relating to creatures, hostile mobs and players themselves.  It likely relates to a creature somewhere in the loaded chunks around you but it could be 100 blocks away from your current position.  Very unlikely that it relates to the clutter blocks etc., those are not animated.

I'll try one more time, covering more of that world and seeing more creatures, staying in the tower probably makes it _less_ likely to happen.

-----

CoB people, when I join the server I get:
* immediate crash on game startup due to a mod, I don't know which, introducing a block called `metalcontainer-north`
* logged startup issue with VanityMod
* warning about a mod improperly using channel `livemap`
* warning about `chiseltools:deeps-880000`
* some texture errors in AncientArmory
* multiple tesselation and texture errors in `CoB_fieldmouse` item
* multiple tesselation and texture errors relating to `decor:witchyitems-1`
* etc.
* FurnitureLibrary exceptions

The first issue makes it unplayable, the game won't start, so I don't understand how people are playing it - I am able to work around that by making a custom build.

**Comment by radfast** [comment]
> While I wouldn't be surprised...there's no Primitive Survival here fyi #4605

I've now been able to reproduce the issue.  On a Primitive Survival **Bioluminescent Sea Creature**, `jelly` variant, the `feed` animation.  Issue occurs in a Creative world with no blocks placed (except some water for it to swim in) and that as the only creature spawned.

I figured this out by starting a new game with the mods AreYSerious sent me (which included Primitive Survival and 75 others), and spawning the non-vanilla creatures, and giving them a bit of time to segue into other animations.

@SpearAndFang I haven't spent any more of my time looking to see _why_ the animation is bugged. In the game code the issue is a missing previous or next keyframe but it's not clear which one (could be either). Hopefully you can figure it out, or ask for help in Discord.

Suffice to say this issue can be closed. I should have gone with my first instincts.

**Comment by SpearAndFang** [comment]
well that's embarrasing. Thanks radfast, I owe you big time.

99% sure the cause is animations (in my case, an idle animation) with no keyframes whatsoever.
@WickedSchnitzel et al new version up on the modDB.

**Comment by WickedSchnitzel** [comment]
> * metalcontainer-north

Can you give me some details about the metalcontainer issue. This must be related to an object that i have extracted from https://www.nexusmods.com/vintagestory/mods/5?tab=files (with permission from the author) using https://mods.vintagestory.at/show/mod/4964#showchangelog
